File: frontend/src/lib/api/crm.ts
import apiClient from './client';

export interface CRMUniversity {
  id: number;
  name: string;
  short_name?: string;
  [key: string]: any;
}

export interface CRMCourse {
  id: number;
  name: string;
  parent_id?: number;
  [key: string]: any;
}

export const crmAPI = {
  // Get universities from CRM
  getUniversities: async (): Promise<CRMUniversity[]> => {
    const response = await apiClient.post<{ success: boolean; data: CRMUniversity[] }>('/crm/universities');
    return response.data.data;
  },

  // Get courses from CRM for a specific university
  getCourses: async (universityId: number): Promise<CRMCourse[]> => {
    const response = await apiClient.post<{ success: boolean; data: CRMCourse[] }>(`/crm/courses?university_id=${universityId}`);
    return response.data.data;
  },
};

File: backend/app/api/routes/crm.py
"""
CRM Routes
Endpoints for fetching master data from Digivarsity CRM
"""
from typing import Optional
from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session
from app.database import get_db
from app.schemas.common import BaseResponse
from app.services.crm_service import CRMService
from app.core.exceptions import AppException
import logging

logger = logging.getLogger(__name__)

router = APIRouter()


@router.post("/universities", response_model=BaseResponse)
async def get_crm_universities(
    db: Session = Depends(get_db),
):
    """
    Get universities from CRM master data
    Uses POST /common/get_master_dd with identifier: ["university"]
    """
    try:
        crm_service = CRMService(db)
        universities = await crm_service.get_master_data(identifier=["university"])
        
        return BaseResponse(
            success=True,
            message="Universities retrieved successfully",
            data=universities
        )
    except AppException as e:
        raise HTTPException(status_code=e.status_code, detail=e.message)
    except Exception as e:
        logger.error(f"Error fetching CRM universities: {str(e)}")
        raise HTTPException(status_code=500, detail=f"Error fetching universities: {str(e)}")


@router.post("/courses", response_model=BaseResponse)
async def get_crm_courses(
    university_id: int = Query(..., description="University ID (parent_id)"),
    db: Session = Depends(get_db),
):
    """
    Get courses from CRM master data for a specific university
    Uses POST /common/get_master_dd with identifier: ["course"], status: 1, parent_id: university_id
    """
    try:
        crm_service = CRMService(db)
        courses = await crm_service.get_master_data(
            identifier=["course"],
            status=1,
            parent_id=university_id
        )
        
        return BaseResponse(
            success=True,
            message="Courses retrieved successfully",
            data=courses
        )
    except AppException as e:
        raise HTTPException(status_code=e.status_code, detail=e.message)
    except Exception as e:
        logger.error(f"Error fetching CRM courses: {str(e)}")
        raise HTTPException(status_code=500, detail=f"Error fetching courses: {str(e)}")

